docs/saltstack.md

```diff
@@ -51,3 +51,20 @@ Example:
 
 {% endmacro %}
 ```
+
+### Issue: Grains are not getting refreshed within the same salt state
+Reason: To apply the state, Salt needs to convert it from YAML format to a data structure. Since it is impossible to parse a YAML file that has Jinja statements inside, Salt needs to parse Jinja first.
+
+In other words, the pipeline looks like this
+
+1. Parse Jinja
+2. Parse the resulting YAML
+3. Interpret the data structure
+4. Apply the state
+
+Solution:
+
+```
+1. Access the grain in a second state
```

Comment:
might be better to mention that the logic should be split into to SLS files and for slots - they are executed on a stage 3, e.g.

>1. split the logic into two SLS files and resolve the grains in the second one
>2. use [slots](https://docs.saltproject.io/en/latest/topics/slots/index.html) which are evaluated on step 3

Reply:
comment is addressed